### PR TITLE
[Merged by Bors] - ET-4519 make sure we never implicitly create a ES mapping

### PR DESCRIPTION
**Pull request by rustonaut**
- added a ES tenant "schema" migration step
- for now that step  only checks if the ES mapping exist, if not
  log and error and push the mapping 

**References:**

- [ET-4519]

[ET-4519]: https://xainag.atlassian.net/browse/ET-4519?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by rustonaut** [comment]
Tow other things we might do:

- currently we create legacy tenant like a normal tenant if it doesn't exist it does create an alias, but we could instead always create for legacy the ES index based on `storage.elastic.index_name` and always create an alias

-  fetch the mapping and check if the expected mapping is a subset of it

File: web-api-db-ctrl/src/elastic.rs [hunk 1]
```diff
@@ -48,35 +51,73 @@ pub(crate) async fn setup_legacy_tenant(
...
+}
+
+async fn does_tenant_index_exist(
+    elastic: &Client,
+    tenant_id: impl AsRef<str>,
+) -> Result<bool, Error> {
     let response = elastic
-        .with_index(default_index)
+        .with_index(tenant_id)
         .request(Method::GET, [], [])
```
Comment (janpetschexain) [inline web-api-db-ctrl/src/elastic.rs:86]: :art: we could reduce this to a head request for less traffic
```suggestion
        .request(Method::HEAD, [], [])
```

File: web-api-db-ctrl/src/elastic.rs [hunk 2]
```diff
@@ -48,35 +51,73 @@ pub(crate) async fn setup_legacy_tenant(
...
+    if !does_tenant_index_exist(elastic, tenant_id).await? {
+        error!(
+            {%tenant_id},
+            "index for tenant doesn't exist, creating a new index"
+        );
+        create_tenant_index(elastic, tenant_id).await?;
+    }
+    //FIXME code to check if the index is a super-set of the expected index
+    //FIXME code for allowing updates to the ES schema, at lest if
+    //      incremental applicable of the change is possible
```
Comment (janpetschexain) [inline web-api-db-ctrl/src/elastic.rs:76]: :art: typos
```suggestion
    //FIXME code for allowing updates to the ES schema, at least if
    //      incremental application of the change is possible
```

File: web-api-db-ctrl/src/postgres.rs [hunk 3]
```diff
@@ -67,13 +67,15 @@ const MIGRATION_LOCK_ID: i64 = 0;
 ///
 /// 3. Concurrently for each tenant a migration of their
 ///    schema will be run (if needed).
-#[instrument(skip(pool, legacy_setup), err)] //TODO log enable_legacy_tenant = legacy_setup.is_some
-pub(super) async fn initialize<F>(
+#[instrument(skip_all, err)] //TODO log enable_legacy_tenant = legacy_setup.is_some
+pub(super) async fn initialize<F, F2>(
```
Comment (janpetschexain) [inline web-api-db-ctrl/src/postgres.rs:71]: :art: for naming consistency i would either name them consecutively or enumerate them all
```suggestion
pub(super) async fn initialize<F, G>(
```
```suggestion
pub(super) async fn initialize<F1, F2>(
```

**Review by janpetschexain — APPROVED** [review]

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**